src/controllers/_authentication_controllers.py

```diff
@@ -21,11 +23,53 @@ def login_handler():
         )
 
         if result.auth is not None:
-            jwt = result.auth.token
+            jwt = result.auth.token  # noqa: F811
             return {"msg": "Login successful", "jwt": jwt}, 200
 
         return {"msg": "Invalid credentials"}, 401
 
     except Exception as e:
         print("SOAP Error:", str(e))
         return {"msg": "Internal error", "error": str(e)}, 500
+
+
+def challenge():
+    try:
+        data = request.json
+
+        if not data:
+            return {"msg": "No JSON data provided in the request"}, 400
+
+        token = data.get("jwt")
+
+        if not token:
+            return {"msg": "Token is missing in JSON data"}, 400
+
+        decoded_token = jwt.decode(token, options={"verify_signature": False})
```

Comment:
The authentication service should be the unique responsible for validating tokens.